processing/src/main/java/io/druid/query/ordering/StringComparators.java

```diff
@@ -61,7 +62,7 @@ public int compare(String s, String s2)
     public int compare(String s, String s2)
     {
       // Avoid conversion to bytes for equal references
-      if(s == s2){
+      if(Objects.equals(s, s2)){
```

Comment:
Is this really better?

For strings that _are_ equal this is faster but for strings that aren't, it should be slower. I'd guess that most comparisons are between strings that are not equal, so the old code should perform better.